internal/pkg/runtime/launcher/util.go

```diff
@@ -46,3 +48,8 @@ func HidepidProc() bool {
 	}
 	return false
 }
+
+// IsCDIDevice checks whether a string is a valid CDI device selector.
+func IsCDIDevice(str string) bool {
```

Comment:
Let's bring this across to `cdi_linux.go` too. The other things here should end up getting factored out into a higher level utils pkg by 4.0.